File: api/flaskr/route/todo.py
from ..service.todo import *
from functools import wraps
from .common import make_common_response
from flask import Flask, request


def register_todo_handler(app: Flask, path_prefix: str) -> Flask:
    @app.route(path_prefix+'/create', methods=['POST'])
    def create_todo():
        user_id = request.user.user_id
        title = request.get_json().get('title', '')
        description = request.get_json().get('description', '')
        create_new_todo(app, user_id, title, description)
        return make_common_response('ok')

    @app.route(path_prefix+'/all', methods=['GET'])
    def list_all_todo():
        user_id = request.user.user_id
        title = request.args.get('title', '')
        is_done = request.args.get('is_done', '')
        start_date = request.args.get('start_date', '')
        end_date = request.args.get('end_date', '')
        todos = get_todos_by_user(
            app, user_id, title, is_done, start_date, end_date)
        return make_common_response(todos)

    @app.route(path_prefix+'/done', methods=['POST'])
    def mark_todo_done():
        todo_id = request.get_json().get('todo_id', '')
        print('------------------------------------------------------mark_todo_done-----------------------------------')
        print(todo_id)
        mark_todo_as_done(app, todo_id)
        return make_common_response('ok')

    @app.route(path_prefix+'/update', methods=['POST'])
    def update_todo():

        todo_id = request.get_json().get('todo_id', '')
        print(todo_id)
        title = request.get_json().get('title', '')
        description = request.get_json().get('description', '')
        deadline = request.get_json().get('deadline', '')
        update_todo_by_id(app, todo_id, title, description, deadline)
        return make_common_response('ok')

    @app.route(path_prefix+'/delete', methods=['POST'])
    def delete_todo():
        todo_id = request.get_json().get('todo_id', '')
        delete_todo_by_id(app, todo_id)
        return make_common_response('ok')

    # @app.route(path_prefix+'/create', methods=['POST'])
    # def create_todo():
    #     title = request.get_json().get('title', '')
    #     description = request.get_json().get('description', '')
    #     deadline = request.get_json().get('deadline', '')
    #     create_todo(app, title, description, deadline)
    #     return make_common_response('ok')

    return app

File: src/backend-web/src/Api/todo.js
import request from "../Service/Request";

export const GetAllTodos = async (title, is_done, start_date, end_date) => {
  return request({
    url: "/api/todo/all",
    method: "get",
    params: { title, is_done, start_date, end_date },
  });
};

export const MarkTodo = async (id) => {
  return request({
    url: "/api/todo/done",
    method: "post",
    data: { todo_id: id },
  });
};

export const updateTodo = async (todoInfo) => {
  return request({
    url: "/api/todo/update",
    method: "post",
    data: todoInfo,
  });
};

export const deleteTodo = async (todo_id) => {
  return request({
    url: "/api/todo/delete",
    method: "post",
    data: { todo_id },
  });
};
